tests/functional/dyn-drv/non-trivial.nix

```diff
@@ -0,0 +1,75 @@
+with import ./config.nix;
+
+builtins.outputOf (mkDerivation {
+  name = "make-derivations.drv";
+
+  requiredSystemFeatures = [ "recursive-nix" ];
+
+  buildCommand = ''
+    set -e
+    set -u
+
+    PATH=${builtins.getEnv "NIX_BIN_DIR"}:$PATH
+
+    export NIX_CONFIG='extra-experimental-features = nix-command ca-derivations dynamic-derivations'
+
+    declare -A deps=(
+      [a]=""
+      [b]="a"
+      [c]="a"
+      [d]="b c"
+      [e]="b c d"
+    )
+
+    # Cannot just literally include this, or Nix will think it is the
+    # *outer* derivation that's trying to refer to itself, and
+    # substitute the string too soon.
+    placeholder=$(nix eval --raw --expr 'builtins.placeholder "out"')
```

Comment:
If this is a common occurrence, I guess we could create placeholders that unwrap into other placeholders.